includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -3437,8 +3457,10 @@ public function update_order_status() {
 
 			$amount                 = $order->get_total();
 			$payment_method_details = false;
+			$is_changing_payment    = isset( $_POST['is_changing_payment'] ) && filter_var( wp_unslash( $_POST['is_changing_payment'] ), FILTER_VALIDATE_BOOLEAN );
+			$this->is_changing_payment_method_for_subscription();
 
-			if ( $amount > 0 ) {
+			if ( $amount > 0 && ! $is_changing_payment ) {
```

Comment:
Avoid entering the pi flow even though orders of type subscription might have amount > 0.